cmd/dev-selectors.go

```diff
@@ -0,0 +1,149 @@
+	if err != nil {
+		cli.ExitWithError("Failed to process subject context keys and values", err)
+	}
+
+	rows := [][]string{}
+	for _, r := range result {
+		rows = append(rows, []string{r.ExternalField, r.ExternalValue})
+	}
+
+	t := cli.NewTabular().Rows(rows...)
+	cli.PrintSuccessTable(cmd, "", t)
+}
+
+func dev_selectorsTest(cmd *cobra.Command, args []string) {
+	h := cli.NewHandler(cmd)
+	defer h.Close()
+
+	flagHelper := cli.NewFlagHelper(cmd)
+	subject := flagHelper.GetRequiredString("subject")
+	contextType := flagHelper.GetRequiredString("type")
+	selectors := flagHelper.GetStringSlice("selectors", selectors, cli.FlagHelperStringSliceOptions{Min: 1})
+
+	var value any
+	if contextType == "json" || contextType == "" {
+		if err := json.Unmarshal([]byte(subject), &value); err != nil {
+			cli.ExitWithError(fmt.Sprintf("Could not unmarshal JSON subject context input: %s", subject), err)
+		}
+	} else if contextType == "jwt" {
+		token, _, err := new(jwt.Parser).ParseUnverified(subject, jwt.MapClaims{})
+		if err != nil {
+			cli.ExitWithError("Failed to parse JWT token", err)
+		}
+
+		if claims, ok := token.Claims.(jwt.MapClaims); ok {
+			value = claims
+		} else {
+			cli.ExitWithError("Failed to get claims from JWT token", nil)
+		}
+	} else {
+		cli.ExitWithError("Invalid subject context type. Must be of type: [json, jwt]", nil)
+	}
+
+	result, err := handlers.TestSubjectContext(value, selectors)
+	if err != nil {
+		cli.ExitWithError("Failed to process subject context keys and values", err)
+	}
+
+	rows := [][]string{}
+	for _, r := range result {
+		rows = append(rows, []string{r.ExternalField, r.ExternalValue})
+	}
+
+	t := cli.NewTabular().Rows(rows...)
+	cli.PrintSuccessTable(cmd, "", t)
+}
+
+func init() {
+	genCmd := man.Docs.GetCommand("dev/selectors/gen",
+		man.WithRun(dev_selectorsGen),
+	)
+	genCmd.Flags().StringP(
+		genCmd.GetDocFlag("subject").Name,
+		genCmd.GetDocFlag("subject").Shorthand,
+		genCmd.GetDocFlag("subject").Default,
+		genCmd.GetDocFlag("subject").Description,
+	)
+	genCmd.Flags().StringP(
+		genCmd.GetDocFlag("type").Name,
+		genCmd.GetDocFlag("type").Shorthand,
+		genCmd.GetDocFlag("type").Default,
+		genCmd.GetDocFlag("type").Description,
+	)
+
+	testCmd := man.Docs.GetCommand("dev/selectors/test",
+		man.WithRun(dev_selectorsTest),
+	)
+	testCmd.Flags().StringP(
+		testCmd.GetDocFlag("subject").Name,
+		testCmd.GetDocFlag("subject").Shorthand,
+		testCmd.GetDocFlag("subject").Default,
+		testCmd.GetDocFlag("subject").Description,
+	)
+	testCmd.Flags().StringP(
+		testCmd.GetDocFlag("type").Name,
+		testCmd.GetDocFlag("type").Shorthand,
+		testCmd.GetDocFlag("type").Default,
+		testCmd.GetDocFlag("type").Description,
+	)
+	testCmd.Flags().StringArrayVarP(
+		&selectors,
+		testCmd.GetDocFlag("selector").Name,
+		testCmd.GetDocFlag("selector").Shorthand,
+		[]string{},
+		testCmd.GetDocFlag("selector").Description,
+	)
+
+	dev_selectorsCmd.AddSubcommands(genCmd, testCmd)
```

Comment:
We offer two ways of doing this. One with a WithSubcommands and the other like this. I had to add this one due to a race condition. 

I'd like to settle on a unified approach to ensure it's easy to onboard new developers.

Reply:
Updated to `WithSubcommands`.